nautobot/circuits/models.py

```diff
@@ -9,17 +10,64 @@
 from nautobot.core.models.generics import BaseModel, OrganizationalModel, PrimaryModel
 from nautobot.utilities.utils import serialize_object
 from .choices import CircuitTerminationSideChoices
-from .querysets import CircuitQuerySet
 
 
 __all__ = (
     "Circuit",
     "CircuitTermination",
     "CircuitType",
     "Provider",
+    "ProviderNetwork",
 )
 
 
+@extras_features(
+    "custom_fields",
+    "custom_links",
+    "custom_validators",
+    "export_templates",
+    "graphql",
+    "relationships",
+    "webhooks",
+)
+class ProviderNetwork(PrimaryModel):
+    name = models.CharField(max_length=100, unique=True)
+    slug = AutoSlugField(populate_from="name")
+    provider = models.ForeignKey(to="circuits.Provider", on_delete=models.PROTECT, related_name="provider_networks")
+    description = models.CharField(max_length=200, blank=True)
+    comments = models.TextField(blank=True)
+
+    csv_headers = [
+        "provider",
+        "name",
+        "slug",
+        "description",
+        "comments",
+    ]
+
+    class Meta:
+        ordering = ("provider", "name")
+        constraints = (
+            models.UniqueConstraint(fields=("provider", "name"), name="circuits_providernetwork_provider_name"),
+        )
+        unique_together = ("provider", "name")
+
+    def __str__(self):
+        return self.name
+
+    def get_absolute_url(self):
+        return reverse("circuits:providernetwork", args=[self.slug])
+
+    def to_csv(self):
+        return (
+            self.provider.name,
+            self.name,
+            self.slug,
+            self.description,
+            self.comments,
+        )
+
```

Comment:
Suggest adding a custom `display` implementation for the REST API:

```suggestion

    @property
    def display(self):
        return f"{self.provider} {self.name}"
```